controller/controller.go

```diff
@@ -391,3 +378,61 @@ func (c *Controller) Identity() identity.Identity {
 func (c *Controller) GetEventDispatcher() event.Dispatcher {
 	return c.eventDispatcher
 }
+
+func (c *Controller) TryInitializeRaftFromBoltDb() error {
+	val, found := c.config.src["db"]
+	if !found {
+		return nil
+	}
+
+	path := fmt.Sprintf("%v", val)
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			return errors.Wrapf(err, "source db not found at [%v]", path)
+		}
+		return errors.Wrapf(err, "invalid db path [%v]", path)
+	}
+
+	return c.InitializeRaftFromBoltDb(path)
+}
+
+func (c *Controller) InitializeRaftFromBoltDb(sourceDbPath string) error {
+	log := pfxlog.Logger()
+
+	log.Info("waiting for raft cluster to settle before syncing raft to database")
+	start := time.Now()
+	for !c.raftController.IsLeader() {
+		time.Sleep(time.Second)
+		if time.Since(start) > time.Second*30 {
+			log.Panic("cannot sync raft to database, as current node is not the leader")
```

Comment:
should this case return an error? Does it get into an infinite loop if it fails to get leadership?

Reply:
Switched it to only require cluster to have a leader, not that local node is leader. It will panic after 30s if the cluster doesn't elect a leader.